'Home' network connection configuration

pkendall64 opened the pull request:
This PR adds functionality to the ESP8255 & ESP32 WiFi modules that allows the user to set a "home" network SSID and Password.

This means that when selecting WIFI update for modules with this configured the ESP will create both an Access Point and connect as a station with named `elrs_tx.local` or `elrs_rx.local` as appropriate if the device can connect to the "home" network. This means that updating via WiFi is a lot friendlier.

I have made some more _quality-of-life_ changes to the code and UI.
- SSID text box with autocomplete and popup so you can select an access point or enter the SSID if known
- Animated icon next to SSID input while searching for access points
- Added a `compile` script that minifies and gzips the source html files and creates the appropriate `.h` files in the source folder so it's easier to keep them in sync
- Adds the tag or branch and short commit hash as version number in the compiled HTML
- Compiles the HTML as part of the build process

jurgelenas commented:
@pkendall64 I just want to suggest using “bash strict mode” in your shell script.

You can read more about it here: http://redsymbol.net/articles/unofficial-bash-strict-mode/

pkendall64 commented:
This PR is well out of date and I have rewritten the compile script in python so it will work on all platforms.
I'll update the PR one we start on the 1.1 release as I've rebased this work on top of the wifi/firmware detect PR.
